anxcloud/resource_dns_record_test.go

```diff
@@ -0,0 +1,72 @@
+package anxcloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/anexia-it/terraform-provider-anxcloud/anxcloud/testutils/environment"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"go.anx.io/go-anxcloud/pkg/utils/test"
+)
+
+func TestAccAnxCloudDNSRecord(t *testing.T) {
+	environment.SkipIfNoEnvironment(t)
+	zoneName := test.RandomHostname() + ".terraform.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		ProviderFactories: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAnxDNSZoneAndRecord(zoneName, 0),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("anxcloud_dns_record.a_record", "zone_name", "0-"+zoneName),
+					resource.TestCheckResourceAttr("anxcloud_dns_record.a_record", "name", "a-record"),
+					resource.TestCheckResourceAttr("anxcloud_dns_record.txt_record", "name", "txt-record"),
+				),
+			},
+			{
+				Config: testAccAnxDNSZoneAndRecord(zoneName, 1),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("anxcloud_dns_record.a_record", "zone_name", "1-"+zoneName),
+					resource.TestCheckResourceAttr("anxcloud_dns_record.a_record", "name", "a-record"),
+					resource.TestCheckResourceAttr("anxcloud_dns_record.txt_record", "name", "txt-record"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAnxDNSZoneAndRecord(zoneNameSuffix string, recordsZoneIndex uint) string {
+	return fmt.Sprintf(`
+	resource "anxcloud_dns_zone" "test_dns_zones" {
+		count = 2
+		name = "${count.index}-%s"
+		is_master = true
+		dns_sec_mode = "unvalidated"
+		admin_email = "admin@terraform.test"
+		refresh = 100
+		retry = 100
+		expire = 1000
+		ttl = 100
+	}
+
+	resource "anxcloud_dns_record" "a_record" {
+		name = "a-record"
+		zone_name = anxcloud_dns_zone.test_dns_zones[%d].name
+		type = "A"
+		rdata = "1.1.1.1"
+		ttl = 300
+	}
+
+	resource "anxcloud_dns_record" "txt_record" {
+		name = "txt-record"
+		zone_name = anxcloud_dns_zone.test_dns_zones[%d].name
+		type = "TXT"
+		rdata = "hello world"
+		ttl = 300
+	}
+	`, zoneNameSuffix, recordsZoneIndex, recordsZoneIndex)
```

Comment:
could explicit argument indexes (`%[1]d`) be useful here, also allowing to pass `recordsZoneIndex` only once?